pkg/migrate/cstor/volume.go

```diff
@@ -0,0 +1,902 @@
+	csiPV.Spec.AccessModes = pvcObj.Spec.AccessModes
+	csiPV.Spec.ClaimRef = &corev1.ObjectReference{
+		APIVersion: pvcObj.APIVersion,
+		Kind:       pvcObj.Kind,
+		Name:       pvcObj.Name,
+		Namespace:  pvcObj.Namespace,
+	}
+	csiPV.Spec.Capacity = corev1.ResourceList{
+		corev1.ResourceStorage: cvObj.Spec.Capacity,
+	}
+	csiPV.Spec.PersistentVolumeSource = corev1.PersistentVolumeSource{
+		CSI: &corev1.CSIPersistentVolumeSource{
+			Driver:       cstorCSIDriver,
+			FSType:       cvObj.Annotations["openebs.io/fs-type"],
+			VolumeHandle: cvObj.Name,
+			VolumeAttributes: map[string]string{
+				"openebs.io/cas-type":                          "cstor",
+				"storage.kubernetes.io/csiProvisionerIdentity": "1574675355213-8081-cstor.csi.openebs.io",
+			},
+		},
+	}
+	csiPV.Spec.PersistentVolumeReclaimPolicy = corev1.PersistentVolumeReclaimDelete
+	if v.StorageClass.ReclaimPolicy != nil {
+		csiPV.Spec.PersistentVolumeReclaimPolicy = *v.StorageClass.ReclaimPolicy
+	}
+	csiPV.Spec.StorageClassName = v.StorageClass.Name
+	csiPV.Spec.VolumeMode = pvcObj.Spec.VolumeMode
+	return csiPV, nil
+}
+
+func (v *VolumeMigrator) createCVC(pvName string) error {
+	var (
+		err    error
+		cvcObj *cstor.CStorVolumeConfig
+		cvObj  *apis.CStorVolume
+	)
+	cvcObj, err = v.OpenebsClientset.CstorV1().CStorVolumeConfigs(v.OpenebsNamespace).
+		Get(pvName, metav1.GetOptions{})
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return err
+	}
+	if k8serrors.IsNotFound(err) {
+		cvObj, err = cv.NewKubeclient().WithNamespace(v.CVNamespace).
+			Get(pvName, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		annotations := map[string]string{
+			"openebs.io/volumeID":      pvName,
+			"openebs.io/volume-policy": pvName,
+		}
+		labels := map[string]string{
+			"openebs.io/cstor-pool-cluster": v.StorageClass.Parameters["cstorPoolCluster"],
+		}
+		finalizer := "cvc.openebs.io/finalizer"
+		cvcObj = cstor.NewCStorVolumeConfig().
+			WithName(cvObj.Name).
+			WithNamespace(v.OpenebsNamespace).
+			WithAnnotations(annotations).
+			WithLabelsNew(labels).
+			WithFinalizer(finalizer)
+		cvcObj.Spec.Capacity = corev1.ResourceList{
+			corev1.ResourceName(corev1.ResourceStorage): cvObj.Spec.Capacity,
+		}
+		cvcObj.Spec.Provision.Capacity = corev1.ResourceList{
+			corev1.ResourceName(corev1.ResourceStorage): cvObj.Spec.Capacity,
+		}
+		cvcObj.Spec.Provision.ReplicaCount, _ = strconv.Atoi(v.StorageClass.Parameters["replicaCount"])
+		cvcObj.Status.Phase = cstor.CStorVolumeConfigPhasePending
+		cvcObj.VersionDetails = cstor.VersionDetails{
+			Desired:     version.Current(),
+			AutoUpgrade: false,
+			Status: cstor.VersionStatus{
+				Current:            version.Current(),
+				DependentsUpgraded: true,
+			},
+		}
+		_, err = v.OpenebsClientset.CstorV1().CStorVolumeConfigs(v.OpenebsNamespace).
+			Create(cvcObj)
+		if err != nil {
+			return err
+		}
+	}
+	klog.Info("Patching the target svc with cvc owner ref")
+	err = v.patchTargetSVCOwnerRef()
```

Comment:
Good to patch owner reference after verifying. 
Reason: If CVC is created and due some reason if CVC is deleted then Service also will be deleted due to owner reference.
(Or)
If we want to have here it self then we have to add point in document.